flexmeasures/data/migrations/versions/f0ee99278f6f_added_flexmodel_to_generic_asset_model.py

```diff
@@ -51,57 +125,167 @@ def upgrade():
     )
     sensors = result.fetchall()
 
-    # Group relevant sensors by generic_asset_id
-    from collections import defaultdict
+    fields_specs = group_sensors_by_field(sensors, conn, generic_asset_table)
 
-    grouped = defaultdict(list)
-
-    for sensor in sensors:
-        attributes = sensor.attributes or {}
-        if "soc-min" in attributes:
-            grouped[sensor.generic_asset_id].append(sensor)
+    # Check for duplicate keys in the grouped sensors
+    validate_for_duplicate_keys(fields_specs)
 
     # Process each group
-    for asset_id, sensors_with_key in grouped.items():
-        if len(sensors_with_key) > 1:
-            raise Exception(
-                f"Multiple sensors with 'soc-min' found for asset_id {asset_id}: {[s.id for s in sensors_with_key]}"
-            )
+    for field_spec in fields_specs:
+        old_name = field_spec["old_field_name"]
+        new_name = field_spec["new_field_name"]
+
+        # Check if the key exist on any asset's attributes and move that into the asset's flex_model
+        # This is to ensure that the flex_model is not empty in the edge case where the asset has no
+        # sensors or has no sensors using the field/key
+        asset_result = conn.execute(
+            sa.select(
+                generic_asset_table.c.id,
+                generic_asset_table.c.attributes,
+                generic_asset_table.c.flex_model,
+            ).where(generic_asset_table.c.attributes[old_name].isnot(None))
+        )
+        affected_assets = asset_result.fetchall()
+
+        for asset in affected_assets:
+            asset_id = asset.id
+            asset_attrs = asset.attributes or {}
+            flex_model_data = asset.flex_model or {}
```

Comment:
`asset.flex_model` is going to be empty (the column was just created). I think you should use this instead:
```suggestion
            flex_model_data = asset.attributes.get("flex-model", {})
```

Reply:
oh thats true, i forgot i needed to creat a `flex_model` key in attributes